### DISCUSSION TITLE
Index generation related issues

### DISCUSSION THREAD

**Discussion by qintian-qe** [oracle/opengrok#4284] [General]
Today, while searching for a keyword in a different branch of code, I found that one branch can be retrieved, while the other branch cannot be retrieved. Is the branch that cannot be retrieved not generating an index or is it caused by other reasons? (There are corresponding files in the source code)

**Comment by vladak** [comment]
This is by design. The indexer only processes what is found in the file system. If you need to index a distinct branch, you have to check it out. See #425.
